packages/beacon-node/src/network/libp2p/index.ts

```diff
@@ -91,14 +90,15 @@ export async function createNodeJsLibp2p(
       ? prometheusMetrics({
           collectDefaultMetrics: false,
           preserveExistingMetrics: true,
-          registry: nodeJsLibp2pOpts.metricsRegistry,
+          // TODO fix type here
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any
+          registry: nodeJsLibp2pOpts.metricsRegistry as any,
```

Comment:
This should not be required types seem to have exact match, it is likely when you started working on this branch that we had an older prom version in Lodestar but it got updated in  https://github.com/ChainSafe/lodestar/pull/6230.

```suggestion
          registry: nodeJsLibp2pOpts.metricsRegistry,
```

Reply:
Cleaned up the type casts and also removed temporary workaround for discv5 metrics added in https://github.com/ChainSafe/lodestar/pull/6201